### ISSUE TITLE
Add support for CNK codes

### ISSUE THREAD

**Issue by thibaultmol** [markusfisch/BinaryEye#493]
In Belgium pretty much all products sold in pharmacies use a CNK code which is a specific MSI coded barcode:
https://www.apb.be/SiteCollectionDocuments/ALL%20PARTNERS%20ACCESS%20DOC/CODES-CNK-DOC/BROCHURE-CNK-EN.pdf

see example:
![Screenshot 2024-09-18 at 23-07-46 Sharp Scan - thibaultmolleman@apotheekmollemandecoster be - Apotheek Molleman De Coster Mail](https://github.com/user-attachments/assets/1191005b-fd37-4be8-9aeb-ea7a7d7352a2)

Most of them use this CNK code. Others (that have unique codes for getting money back from health insurance have datamatrix codes which already work properly) and a smaller number of products use regular EAN codes which also work.

**Comment by markusfisch** [comment]
Thank you for bringing this up! I've never seen these barcodes, unfortunately. Very interesting! And sorry for the slow response 🙈 

Now, this barcode format would have to be added to [ZXingCpp](https://github.com/zxing-cpp/zxing-cpp) (that's the library this app uses to read barcodes). @axxel Do you know CNK codes?

**Comment by axxel** [comment]
No, I haven't seen these before. The link to the PDF seems broken. @thibaultmol can you double check that?

**Comment by thibaultmol** [comment]
XD timing. They (APB) literally changed to a new website TODAY xd amazing

https://web.archive.org/web/20240808102218/https://www.apb.be/SiteCollectionDocuments/ALL%20PARTNERS%20ACCESS%20DOC/CODES-CNK-DOC/BROCHURE-CNK-EN.pdf

**Comment by axxel** [comment]
I see. They use "just" a fixed length [MSI](https://en.wikipedia.org/wiki/MSI_Barcode) code. That symbology is pretty simple in its construction. I guess this could be implemented along the lines of the existing [Code39 reader](https://github.com/zxing-cpp/zxing-cpp/blob/master/core/src/oned/ODCode39Reader.cpp).

That said, I don't have spare time to put into this right now. But if there is a business interest (money involved), my priorities can be influenced :). In that case, please get in touch via zxingcpp at gmail dot com.
